numpy/core/src/multiarray/lowlevel_strided_loops.c.src

```diff
@@ -1724,10 +1738,20 @@ mapiter_@name@(PyArrayMapIterObject *mit)
                         NPY_AUXDATA_FREE(transferdata);
                         return -1;
                     }
+                    if (is_reset_trivial != 0) {
+                        /* resetting is trivial, calculate the offsets */
+                        reset_offsets[0] = subspace_ptrs[0] - self_ptr;
+                        reset_offsets[1] = subspace_ptrs[1] - mit->extra_op_ptrs[0];
+                        /* use the faster adjustment */
+                        is_reset_trivial ++;
+                    }
                 }
                 else {
-                    subspace_ptrs[0] = self_ptr;
-                    subspace_ptrs[1] = mit->extra_op_ptrs[0];
+                    /*
+                     * will avoid resetting if the reset is trival.
```

Comment:
Possibly could try to expand this a little along the same lines as above, especially an example might help. If the inner/subspace iterator is trivially iterable, we still may need to calculate the correct starting point of the iteration. The offset is typically zero unless reversing the iteration is more efficient (originally a negative stride)  in reversed order in which case this is the offset to the last item.